lib/news_scraper/extractors/google_news_rss.rb

```diff
@@ -39,18 +37,18 @@ def google_urls_from_resp(body)
         end
       end
 
-      def extract_article_uris(google_urls)
+      def extract_article_urls(google_urls)
         google_urls.map do |google_url|
-          regex = google_url.match(/&url=https?:\/\/(.*)/)
-          regex.nil? ? nil : regex[1]
+          regex = google_url.match(%r(&url=(?<url>https?://.*)))
+          regex.nil? ? nil : regex['url']
         end.compact.uniq
       end
 
-      def write_to_temp(article_uris)
-        filename = "#{query.downcase.gsub(/\s/, '_')}_#{Time.now.to_i}.txt"
-        puts "Writing article uris to #{filename}"
-        File.open(File.join(TMP_DIR, filename), 'w') do |file|
-          article_uris.each { |uri| file.puts uri }
+      def write_to_temp(article_urls)
+        filename = "#{@query.downcase.gsub(/\s/, '_')}_#{Time.now.to_i}.txt"
```

Comment:
As stated above, this was for a means to log the article urls in case
Google decided to 🔥 us.

On Sep 1, 2016 8:15 PM, "Julian Nadeau" <notifications@github.com> wrote:

> In lib/news_scraper/extractors/google_news_rss.rb
> <https://github.com/richardwu/news_scraper/pull/6#discussion_r77276429>:
>
> >          end.compact.uniq
> >        end
> >
> > -      def write_to_temp(article_uris)
> > -        filename = "#{query.downcase.gsub(/\s/, '_')}_#{Time.now.to_i}.txt"
> > -        puts "Writing article uris to #{filename}"
> > -        File.open(File.join(TMP_DIR, filename), 'w') do |file|
> > -          article_uris.each { |uri| file.puts uri }
> > +      def write_to_temp(article_urls)
> > +        filename = "#{@query.downcase.gsub(/\s/, '_')}_#{Time.now.to_i}.txt"
>
> Why do we need time here if its in a tmp dir?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/richardwu/news_scraper/pull/6/files/1fff18f3aa533a4151a86d9d3bbaa3f0cb80189d#r77276429>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AKEu8sa8xM1KwRkY-k32eyUA1Sc2bvIlks5ql2q9gaJpZM4JzP8F>
> .
>